crates/store/src/state.rs

```diff
@@ -946,86 +966,88 @@ impl State {
             let witness = inner_state.account_tree.open_latest(account_id);
             (block_num, witness)
         };
-        drop(inner_state);
 
-        let account_details = if let Some(AccountDetailRequest {
+        Ok((block_num, witness))
+    }
+
+    /// Fetches the account details (code, vault, storage) for a public account at the specified
+    /// block.
+    ///
+    /// This method queries the database to fetch the account state and processes the detail
+    /// request to return only the requested information.
+    async fn fetch_account_proof_details(
+        &self,
+        account_id: AccountId,
+        block_num: BlockNumber,
+        detail_request: AccountDetailRequest,
+    ) -> Result<AccountDetails, DatabaseError> {
+        let AccountDetailRequest {
             code_commitment,
             asset_vault_commitment,
             storage_requests,
-        }) = details
-        {
-            let account_info = self.db.select_historical_account_at(account_id, block_num).await?;
-
-            // if we get a query for a _private_ account _with_ details requested, we'll error out
-            let Some(account) = account_info.details else {
-                return Err(DatabaseError::AccountNotPublic(account_id));
-            };
+        } = detail_request;
 
-            let storage_header = account.storage().to_header();
+        let account_info = self.db.select_historical_account_at(account_id, block_num).await?;
 
-            let mut storage_map_details =
-                Vec::<AccountStorageMapDetails>::with_capacity(storage_requests.len());
+        // if we get a query for a _private_ account _with_ details requested, we'll error out
+        let Some(account) = account_info.details else {
+            return Err(DatabaseError::AccountNotPublic(account_id));
+        };
```

Comment:
If for some reason we corrupt or screw up the database such that a public account is missing the details then this is going to be quite confusing.

Maybe we can rename to `fn fetch_public_account_proof_details` and make this a `panic!` instead, pushing the invariant to the caller (which is our pub fn).

A more type safe approach could be wrapping `AccountDetailRequest` such that the new type can only be constructed if the account ID is public.